samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -620,3 +625,217 @@ def _link_lambda_function_to_layer(
+
+    tf_layers: Dict[str, Dict]
+        Dictionary of all actual terraform layers resources (not configuration resources). The dictionary's key is the
+        calculated logical id for each resource
+
+    Returns
+    --------
+    List:
+        The list of layers after processing
+
+    """
+    LOG.debug(
+        "Map the resolved layers %s to configuration function %s.", resolved_layers, function_tf_resource.full_address
+    )
+    layers = []
+    for resolved_layer in resolved_layers:
+        if isinstance(resolved_layer, ConstantValue):
+            layers += _process_constant_layer_value(function_tf_resource, resolved_layer)
+        else:
+            layers += _process_reference_layer_value(function_tf_resource, resolved_layer, tf_layers)
+    return layers
+
+
+def _process_constant_layer_value(function_tf_resource: TFResource, resolved_layer: ConstantValue) -> List:
+    """
+    Process the layer value of type ConstantValue.
+
+    Parameters
+    ----------
+    function_tf_resource: TFResource
+        The input lambda function terraform configuration resource
+
+    resolved_layer: ConstantValue
+        The layer arn value.
+
+    Returns
+    -------
+    List
+        The resolved layers values that will be used as a value for the mapped CFN function Layers attribute.
+    """
+    LOG.debug("Process the constant layer %s.", resolved_layer.value)
+    layer_arn = resolved_layer.value
+    if not isinstance(layer_arn, str):
+        LOG.debug("The constant layer %s is not a string.", resolved_layer.value)
+        raise InvalidResourceLinkingException(
+            f"Could not use the value {layer_arn} as a Layer for lambda function "
+            f"{function_tf_resource.full_address}. Lambda Function Layer value should be a valid ARN String"
+        )
+    return [layer_arn]
+
+
+def _process_reference_layer_value(
+    function_tf_resource: TFResource, resolved_layer: ResolvedReference, tf_layers: Dict[str, Dict]
+) -> List:
+    """
+    Process the layer value of type ResolvedReference.
+
+    Parameters
+    ----------
+    function_tf_resource: TFResource
+        The input lambda function terraform configuration resource
+
+    resolved_layer: ResolvedReference
+        The layer reference.
+
+    tf_layers: Dict[str, Dict]
+        Dictionary of all actual terraform layers resources (not configuration resources). The dictionary's key is the
+        calculated logical id for each resource
+
+    Returns
+    -------
+    List
+        The resolved layers values that will be used as a value for the mapped CFN function Layers attribute.
+
+    """
+    LOG.debug("Process the reference layer %s.", resolved_layer.value)
+    # skip processing the layers defined using the data source block, as it should be mapped to
+    # Layer ARN string while executing the terraform plan command.
+    if resolved_layer.value.startswith("data.aws_lambda_layer_version."):
```

Comment:
we should move these string constants elsewhere.

Reply:
done